code/modules/mob/living/bot/bot.dm

```diff
@@ -296,13 +296,16 @@
 	return 1
 
 /mob/living/bot/proc/handlePatrol()
-	makeStep(patrol_path)
+	if(makeStep(patrol_path) && frustration)
+		frustration = 0
+	else if(max_frustration)
+		++frustration
```

Comment:
Shouldn't this make sure we don't go over the max, or are we just delegating that to some other system elsewhere?

Reply:
This is a mirror of the frustration code used for target pathfinding on L277. I believe the idea is if max_frustration is not set a value (ie you don't want any frustration logic to apply to a bot), then frustration is never counted in the first place and so doesn't trigger frustration checks elsewhere in the code.

max_frustration is handled on L242 and L252 for target pathfinding and patrol pathfinding respectively, with frustration being reset upon a single successful step taken by the bot (L275 & L300)